lib/chitter_manager.rb

```diff
@@ -0,0 +1,38 @@
+require 'pg'
+
+class ChitterManager
+
+  attr_reader :id, :message, :date
+
+  def initialize(id:, message:, date:)
+    @id = id
+    @message = message
+    @date = date
+  end
+
+  def self.all
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else 
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec("SELECT * FROM peeps ORDER BY peep_date DESC;")
+    result.map do |peep| 
+      ChitterManager.new(id: peep['id'], message: peep['message'], date: peep['peep_date'])
```

Comment:
It's a bit strange that `ChitterManager` is also used to wrap the data for a single peep. I would expect a separate class for that, e.g. `Peep`, or instead, that `ChitterManager` is renamed to `Peep`. 

In software, calling a class `Manager` is often taken to imply that it manages access or controls a bunch of functionality (and your class does this well with the `all` and `post` methods). This means that implicitly we expect there to only ever be one manager, not one manager per peep.

Reply:
Thanks I will refactor this to make it more coherent.